File: JobRecruitment/src/main/java/bachelor/thesis/job_recruitment/core/model/Contractor.java
package bachelor.thesis.job_recruitment.core.model;

import lombok.*;

import javax.persistence.*;
import java.util.List;

@EqualsAndHashCode(callSuper = true)
@Data
@AllArgsConstructor
@NoArgsConstructor
@Entity
@Builder
public class Contractor extends BaseEntity<Long>{
    @Column(length = 50)
    private String name;
    @Column(length = 3000)
    private String description;
    private Integer nrOfEmployees;


    @OneToOne(cascade = CascadeType.ALL, fetch = FetchType.EAGER, orphanRemoval = true)
    private File logo;

    @OneToOne(cascade = CascadeType.ALL, fetch = FetchType.EAGER, orphanRemoval = true)
    private Location location;
    @OneToOne(fetch = FetchType.EAGER)
    private GenericUser owner;

    /*@Override
    public String toString(){
        return "contractor";
    }*/

}

File: JobRecruitment_Frontend/src/app/model/Contractor.ts
import {BaseEntity} from "./BaseEntity";
import {Location} from "./Location";
import {Job} from "./Job";
import {FileProperties} from "./FileProperties";
import {User} from "./User";

export class Contractor extends BaseEntity{
  name : string;
  description : string;
  nrOfEmployees: number;
  logo : FileProperties;
  location : Location;
  owner: User;


  constructor(id: number, name: string, description: string, nrOfEmployees: number, logo: FileProperties, location: Location, owner: User) {
    super(id);
    this.name = name;
    this.description = description;
    this.nrOfEmployees = nrOfEmployees;
    this.logo = logo;
    this.location = location;
    this.owner = owner;
  }
}
